ifaddr/_posix.py

```diff
@@ -62,18 +66,20 @@ def add_ip(adapter_name, ip):
 
     while addr:
         name = addr[0].ifa_name.decode(encoding='UTF-8')
-        ip = shared.sockaddr_to_ip(addr[0].ifa_addr)
-        if ip:
+        ip_addr = shared.sockaddr_to_ip(addr[0].ifa_addr)
+        if ip_addr:
             if addr[0].ifa_netmask and not addr[0].ifa_netmask[0].sa_familiy:
                 addr[0].ifa_netmask[0].sa_familiy = addr[0].ifa_addr[0].sa_familiy
             netmask = shared.sockaddr_to_ip(addr[0].ifa_netmask)
             if isinstance(netmask, tuple):
                 netmaskStr = str(netmask[0])
                 prefixlen = shared.ipv6_prefixlength(ipaddress.IPv6Address(netmaskStr))
-            else:
+            elif isinstance(netmask, str):
```

Comment:
Please don't change runtime behavior (even theoretically). If you have some mypy-related issues in this block use casts (if in doubt let me know).

Reply:
I've changed this back, however, the non-existing error handling is an issue (which can be fixed later I guess).

Reply:
Sure, if there's anything missing here I'm happy to accept a patch improving things. I don't think we can get `None` here but maybe I'm wrong.